File: CS/PS/Site/programmers/lv2/짝지어제거하기.js
function solution(s) {
  var answer = 0;
  let len = s.length;
  let stk = [];
  for (let i = 0; i < len; i++) {
    let top = stk.length - 1;
    if (stk[top] === s[i]) {
      stk.pop();
    } else {
      stk.push(s[i]);
    }
  }
  return stk.length >= 1 ? 0 : 1;
}

console.log(solution("baabaa"));

File: CS/PS/Site/programmers/python/lv2/짝지어제거하기.py
def solution(s):
    lst = list(s)
    stk = [lst[0]]
    for v in range(1, len(lst)):
        if (stk and stk[-1] == lst[v]):
            stk.pop()
        else :
            stk.append(lst[v])
    return 1 if len(stk) == 0 else 0